enhancements/console/pinned-resources.md

```diff
@@ -0,0 +1,190 @@
+
+Some customers want to provide defaults for those pre-pinned resources for all users.
+
+On the Developer perspective navigation, users can see the default pinned resources if they have not customized these pinned resources. Cluster admins have no way to set these default pinned resources for their users. Users can change their pinned resources on their own. The user-pinned resources are saved in the user settings (as-per user-settings ConfigMap).
+
+This enhancement allows cluster admins to define pre-pinned resources for new users and users who have not customized their navigation items. Users should still be able to change these defaults and customize the pinned resources on their own. New changes from a cluster admin after a user customize their own pinned resources do not have to be shown.
+
+![Default pinned resources](default-pinned-resources.png)
+
+## Motivation
+
+Some customers want to provide default pre-pinned resources for all users.
+
+### User Stories
+
+#### Story 1
+
+Customer wants to customize the developer perspective navigation by providing pre-pinned resources to users who do not have their navigation items customized.
+
+As a cluster admin, I want to provide default pre-pinned resources for all users so that the developers who donot have their navigation items customized will see these default pinned resources.
+
+### Goals
+
+- Allow cluster admins to define default pre-pinned resources on the dev perspective navigation
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+We can extend the existing `operator.openshift.io/v1` / `Console CRD`.
+We can add `spec.customization.perspectives.pinnedResources` which lists default pre-pinned resources
+
+### Workflow Description
+
+To add default pre-pinned resources:
+
+```text
+├── spec
+│   ├── customization
+|       ├── perspectives
+|           ├── id
+|           ├── ...
+|           ├── pinnedResources
+|               └── group
+                └── version
+                └── resource
+└── ...
+```
+
+Each perspective conforms to the following schema :
+
+- `pinnedResources: operatorv1.PinnedResourceReference[]` , defines a list of pinned resources. The default pinned resources are contributed by the console UI extension plugin.  
+- `group: string` , defines the group of the resource.
+- `version: string` , defines the version of the resource.
+- `resource: string` , defines the type of the resource. It is the plural form of the resource kind in lowercase.
+
+If the `pinnedResources` list is empty, the default pinned resources contributed by the console UI extension plugin will be shown.
+If the `pinnedResources` list is non-empty, only the listed items will be shown if the user hasn't already customized the navigation items.
+If the `group`, `version` or `resource` is incorrect, the resource to be pinned will be ignored.
```

Comment:
What do you mean by "incorrect"? Do you mean the resource doesn't exist on the cluster?

Have we considered how this will work with multi-cluster in 4.13 where console talks to different clusters that might have different sets of resources?

Reply:
Pinned resources aren't shown when the CRD (model) isn't found. So switching should not be an issue. 

And isn't the customization configuration per cluster? Is the index and Server flags reloaded when Switching a cluster?

We should def. test this.

Reply:
When switching between clusters the invalid resources should disappear. We will test this

Reply:
The console customization comes from the hub since that is where the console is running. We don't read console configuration from the spoke clusters.

Reply:
I would clarify in the enhancement what you mean by "incorrect" here.